src/workerd/io/worker-fs.c++

```diff
@@ -1738,6 +1796,46 @@ class StdioFile final: public File, public kj::EnableAddRefToThis<StdioFile> {
  private:
   VirtualFileSystem::Stdio type;
   mutable kj::Maybe<kj::String> maybeUniqueId;
+
+  static constexpr size_t MAX_LINE_BUFFER_SIZE = 4096;
+  static constexpr size_t MAX_WRITE_SIZE = 16 * 1024;
+  mutable kj::Vector<kj::byte> stdoutLineBuffer;
+  mutable kj::Vector<kj::byte> stderrLineBuffer;
+  mutable bool microtaskScheduled = false;
+
+  kj::Rc<WeakRef<StdioFile>> weakThis;
+
+  kj::Vector<kj::byte>& getLineBuffer() const {
+    return (type == VirtualFileSystem::Stdio::OUT) ? stdoutLineBuffer : stderrLineBuffer;
+  }
+
+  void scheduleFlushMicrotask(jsg::Lock& js) {
+    if (microtaskScheduled) return;
+    microtaskScheduled = true;
+
+    // Create ephemeral callback with weak reference for safety
+    auto callback = js.wrapSimpleFunction(js.v8Context(),
+        [weakThis = weakThis->addRef()](jsg::Lock& js, const v8::FunctionCallbackInfo<v8::Value>&) {
+      weakThis->runIfAlive([&](StdioFile& self) {
+        self.microtaskScheduled = false;
+
+        if (self.stdoutLineBuffer.size() > 0) {
+          if (IoContext::hasCurrent()) {
```

Comment:
Just to simplify things slightly, we can probably get by with just a single `IoContext::hasCurrent()` check and do both writes within it/